serverv2/core/transaction/validation.go

```diff
@@ -0,0 +1,19 @@
+package transaction
+
+import "context"
+
+type Tx interface {
+	Hash() [32]byte // TODO evaluate if 32 bytes is the right size & benchmark overhead of hashing instead of using identifier
+}
+
+// Handler is a recursive function that takes a transaction and returns a new context to be used by the next handler
+type Handler func(ctx context.Context, tx Tx, simulate bool) (newCtx context.Context, err error)
```

Comment:
I would get rid of `simulate` and assume exec mode can be extracted from the context.Context